src/main/java/nl/knaw/huc/resussun/dataextension/DataExtensionClient.java

```diff
@@ -30,40 +31,68 @@ public DataExtensionClient() {
+                                                            .filter(prop -> propertyIds.contains(prop.getName()))
+                                                            .collect(Collectors.toList());
 
       TimbuctooExtensionQuery queryCreator = new TimbuctooExtensionQuery(
           api.getDataSourceId(),
           extensionRequest.getEntityIds(),
-          propertyIds
+          properties
       );
       TimbuctooRequest timbuctooRequest = queryCreator.createQuery();
       rows = timbuctoo.executeRequest(timbuctooRequest, queryCreator.createMapper());
     }
 
-    List<ResponseExtendProperty> properties = extensionRequest.getProperties().stream().map(prop -> {
-      return new ResponseExtendProperty(
-          prop.getId(),
-          propertyIdNameMap.getOrDefault(prop.getId(), ""),
-          prop.getSettings().orElse(null)
-      );
+    final Map<String, PropertyMetadata> propertyIdPropertyMap = createPropertyIdPropertyMap(dataSetMetadataMap);
+    List<DataExtensionResponsePropertyMetadata> properties = extensionRequest.getProperties().stream().map(prop -> {
+      final PropertyMetadata metadata = propertyIdPropertyMap.getOrDefault(prop.getId(), null);
+
+      if (metadata.isValueType()) {
+        return DataExtensionResponse.createPropertyMetadata(
+            prop.getId(),
+            metadata.getUri(),
+            prop.getSettings().orElse(null)
+        );
+      } else {
+        return DataExtensionResponse.createPropertyMetadata(
+            prop.getId(),
+            metadata.getUri(),
+            prop.getSettings().orElse(null),
+            getType(metadata, dataSetMetadataMap)
+        );
+      }
     }).collect(Collectors.toList());
     return new ObjectMapper().valueToTree(new DataExtensionResponse(properties, rows));
   }
 
-  private Map<String, String> createPropertyIdNameMap(ApiData api, Timbuctoo timbuctoo)
-      throws TimbuctooException {
-    return timbuctoo.executeRequest(createCollectionsMetadataRequest(api.getDataSourceId()), collectionIdAsKey())
-                    .entrySet().stream()
-                    .flatMap((entry) -> entry.getValue().getProperties().stream())
-                    .filter(prop -> !prop.getName().startsWith("_inverse_"))
-                    .distinct()
-                    .collect(Collectors.toMap(PropertyMetadata::getUri, PropertyMetadata::getName));
+  private DataExtensionResponse.Type getType(PropertyMetadata metadata,
+                                             Map<String, CollectionMetadata> dataSetMetadataMap) {
+    final List<String> refColIds = metadata.getReferencedCollectionIds();
+
+    return dataSetMetadataMap.entrySet().stream()
+                             .filter(entry -> refColIds.contains(entry.getKey()))
+                             .map(Map.Entry::getValue)
+                             .map(col -> new DataExtensionResponse.Type(col.getUri(), col.getUri()))
```

Comment:
The type has an id (the uri) and a name (the collection id). So the mapping should be:
`.map(col -> new DataExtensionResponse.Type(col.getCollectionId(), col.getUri()))`

Reply:
The same should happen in the `SearchClient`, line 105:
```
// Types are mapped to a list of strings in ElasticSearch, so we can safely cast the object to a list
final List<String> types = (List<String>) source.get("types");
types.forEach(type -> candidate.type(type, type));
```

Somewhere, we should zip the types with the collection ids I guess?
```
final List<String> collectionIds = (List<String>) source.get("collectionIds");
```